File: Sorting and Searching/Missing Coin Sum/missing_coin_sum.cpp
#include<bits/stdc++.h>
using namespace std;
#define ll long long int

void solve(){
	ll n, temp, sum = 0, ans;
	cin>>n;
	vector<int> arr(n);

	for(int i=0;i<n;i++){
		cin>>temp;
		arr[i] = temp;
	}

	sort(arr.begin(), arr.end());

	int i=0;
	while(i<n && arr[i]<=sum+1){
		sum+=arr[i];
		i++;
	}
	cout<<sum+1;
    return;
	
}

int main(){
	long long testCases;
	testCases = 1;
	// cin>>testCases;
	while(testCases--){
		solve();
	}
	return 0;
}

File: Sorting and Searching/Missing Coin Sum/missingCoinSum.java
import java.util.Arrays;
import java.util.Scanner;

public class missingCoinSum {
    private static void solve(){
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        long sum = 0;
        int[] arr = new int[n];
        for(int i=0;i<n;i++){
            arr[i] = scanner.nextInt();
        }
        Arrays.sort(arr);
        for(int i=0;i<n && arr[i] <= sum+1; i++){
            sum += arr[i];
        }
        System.out.println(sum+1);
    }
    public static void main(String args[]){
        solve();
    }
}
